### PR TITLE
Typescriptify VModal components

### LINKED ISSUE #1925: Add types to VModal components
## Description

Add type checking to components inside `VModal`. Make sure to add the directory to tsconfig.json's include list.

Additional context
The `VModalContent` imports an SVG logo converting it to a component with `?inline`, which cannot be type-checked. It should be replaced with an `svg` element similar to how it's used in `VIcon`.
Part of an ongoing effort to add type checking to parts of the project that can be type checked.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1925 by @obulat

## Description

This PR converts JSDoc types to TS syntax, and adds the `VModal` directory to `tsconfig.json`

## Testing Instructions

The `types` step of CI should pass.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VHeaderOld/VLogoButtonOld.vue [hunk 1]
```diff
@@ -29,11 +32,11 @@ import { defineComponent } from '@nuxtjs/composition-api'
 import VLogoLoader from '~/components/VLogoLoader/VLogoLoader.vue'
 import VButton from '~/components/VButton.vue'
 
-import OpenverseLogoText from '~/assets/icons/openverse-logo-text.svg?inline'
+import OpenverseLogoText from '~/assets/icons/openverse-logo-text.svg'
```
Comment (dhruvkb) [inline src/components/VHeaderOld/VLogoButtonOld.vue:35]: I don't remember if there was a specific reason we are getting rid of `?inline`. But this is leading to the SVG tag being duplicated in both the SVG file and in the HTML.

File: tsconfig.json [hunk 2]
```diff
@@ -85,7 +85,7 @@
     "src/components/VHeader/VPageLinks.vue",
     "src/components/VHeader/VSearchBar/**.vue",
     "src/components/VImageDetails/VRelatedImages.vue",
-    "src/components/VModal/VInputModal.vue",
+    "src/components/VModal/**.vue",
```
Comment (dhruvkb) [inline tsconfig.json:88]: `**` are in the upper block.

**Review by dhruvkb — APPROVED** [review]
No blocking comments, but 1 nit and a 1 question.

File: src/components/VHeaderOld/VLogoButtonOld.vue (hunk 1 above)
Comment (obulat) [reply src/components/VHeaderOld/VLogoButtonOld.vue:35]: I wrote about the reasons in this issue: https://github.com/WordPress/openverse-frontend/issues/1715.
I haven't noticed the duplication, I'll look into it, thank you!

Comment (obulat) [reply src/components/VHeaderOld/VLogoButtonOld.vue:35]: I don't see the `SVG` tag duplication, @dhruvkb:
Screen Shot 2022-10-27 at 9 23 40 PM

File: tsconfig.json (hunk 2 above)
Comment (obulat) [reply tsconfig.json:88]: Great catch :)

**Review by sarayourfriend — APPROVED** [review]
LGTM! Glad to see this one finally converted. Nice work getting through all of its dependencies :partying_face:

**Pull request merged**